Gen_Extract copies instead of removing from the results list

TamaMcGlinn opened the pull request:
Gen_Extract now copies into the list given without clobbering the existing results list. This makes it possible to generate two reports in sequence. Without this you would get incorrect output if you did that:

```
Total Tests Run:   1
Successful Tests:  0
Failed Assertions: 0
Unexpected Errors: 0
Cumulative Time: 0.000004 sec. seconds
```

This is because the total tally is simply an integer field that is not updated, while the actual lists that are consulted when asking how many successful tests there were are destroyed when you try to retrieve the succeeded, failed or errored tests.

The comments around the functions still need to be updated to remove the note about it 'resetting' lists.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AdaCore/aunit?pullRequest=7) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/AdaCore/aunit?pullRequest=7) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AdaCore/aunit?pullRequest=7) it.</sub>

TamaMcGlinn commented:
This fixes #6 Test_Results get functions modify state

TamaMcGlinn commented:
I rebased, removed the indicated obsolete comments, and also fixed the signatures of those functions; they are `in` rather than in out to reflect the fact that the lists passed in aren't changed.

fedor-rybin commented on include/aunit/framework/aunit-test_results.adb:
```diff
@@ -60,7 +60,7 @@ package body AUnit.Test_Results is
       with function Test (Position : Result_Lists.Cursor) return Boolean;
    procedure Gen_Extract (R : in out Result;
-                          E : in out Result_Lists.List);
+                          E : out Result_Lists.List);
```
I'm a bit confused. In the log you say that you are making the parameter in instead of out, but here it is now out instead of in.
I belive what you indended to to do was make the **first** parameter in instead ouf in out?

TamaMcGlinn replied:
Quite right, I intended to change the private result object, not the list.

TamaMcGlinn commented:
@fedor-rybin Could you have another look?

fedor-rybin commented:
This is now merged. Thank you for your contrubution and your patience!
